i16.go

```diff
@@ -1,13 +1,24 @@
 package goscale
 
-import "bytes"
+import (
+	"bytes"
+	"math/big"
+)
 
 type I16 int16
 
 func (value I16) Encode(buffer *bytes.Buffer) error {
 	return U16(value).Encode(buffer)
 }
 
+func NewI16(n int16) I16 {
+	return I16(n)
+}
+
+func (value I16) ToBigInt() *big.Int {
```

Comment:
If you implemented `ToBigInt()` solely to enable passing the types to the generic function, you can replace the interface with a constraint, eliminating the need to implement `ToBigInt()` for each type. 

Instead of using an interface:
```go
type BigNumbers interface {
	ToBigInt() *big.Int
}
```
you can use an interface constraint:
```go
type BigNumbers interface {
	U8 | U16 | U32 | U64 | U128 | I8 | I16 | I32 | I64 | I128
}
```
this avoids the need to implement `ToBigInt()` for each type.

Reply:
never mind, you are actually using the `ToBigInt()` method elsewhere